File: app/Market/h/src/Pages/Profits.hs
-- Copyright 27-Dic-2018 ºDeme
-- GNU General Public License - V3 <http://www.gnu.org/licenses/>

-- | Profits page

module Pages.Profits (process) where

import Data.List
import Text.Printf
import qualified Dm.Js as Js
import Dm.Js (JSValue)
import qualified Dm.Cgi as Cgi
import Dm.Cgi (Cgi)
import qualified Dm.File as File
import qualified Data.Historic as Historic
import qualified Global as G

process :: Cgi -> [(String, JSValue)] -> IO ()
process cgi rq = do
  case Cgi.get rq Js.rString "rq" of
    "idata" -> do
      d <- Historic.take 0
      Cgi.ok cgi [("data", Js.wList d)]
    "removeDuplicate" -> do
      Historic.removeDuplicate
      Cgi.empty cgi
    s -> error $ printf "Unknown rq '%s'" s

File: app/MultiMarket/go/server/pgs/acc/profits/profits.go
// Copyright 19-Sep-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Daily charts page.
package profits

import (
	"fmt"
	"github.com/dedeme/MultiMarket/db/acc/profitsDb"
	"github.com/dedeme/MultiMarket/global/sync"
	"github.com/dedeme/golib/cgi"
	"github.com/dedeme/golib/json"
)

func Process(ck string, mrq map[string]json.T) string {
	rq := cgi.RqString(mrq, "rq")
	switch rq {
	case "idata":
		rp := map[string]json.T{}
		sync.Run(func(lk sync.T) {
			rp["data"] = profitsDb.Read(lk)
		})
		return cgi.Rp(ck, rp)
	default:
		panic(fmt.Sprintf("Value of rq ('%v') is not valid", rq))
	}
}
